### PR TITLE
Grouped data by city and calculated overall baseline averages for PM10 and PM25

### PR DESCRIPTION
**Pull request by tejasph**
Clustered Station locations based on their cities, which narrowed unique locations to 55. Altered process_dataset function to also calculate ovrall averages for PM10 and PM25. Finally, rearranged code to make it more readable.

I have added a new dataframe (processed_baseline_data.csv). The reason I included it as a separate DataFrame is because the baseline visual will most likely be a permanent fixture and shouldn't be part of the dropdown menu options. However, if needed I can adjust the code to include the baseline data as part of the processed DataFrame.

If there's time, I will clean the code further.

### CODE REVIEW

**Comment by KarlosMuradyan** [comment]
Looks great!! I think we will need that averages more than we thought.  The lines in the lineplot of all locations and one PM are too much, so most likely either we need to put only average of that specific PM and highlighted locations. Attached, for example you can see lineplot with only 10 locations, but it is still annoying. You can imagine how annoying will be to have all locations in the same plot.

In short great idea to have average!!
